build-tools/rpm/macros.yast

```diff
@@ -15,49 +15,85 @@
 %yast_controldir %{yast_dir}/control
 %yast_schemadir %{yast_dir}/schema
 %yast_scrconfdir %{yast_dir}/scrconf
-%yast_desktopdir %{_prefix}/share/applications/YaST2
+%yast_desktopdir %{_datadir}/applications/YaST2
 
 %yast_ybindir %{_prefix}/lib/YaST2/bin
 %yast_ystartupdir %{_prefix}/lib/YaST2
-%yast_plugindir %{_libdir}/YaST2/plugin
-%yast_includedir %{_prefix}/include/YaST2
-%yast_docdir %{_prefix}/share/doc/packages/%{name}
-%yast_logdir /var/log/YaST2
-%yast_vardir /var/lib/YaST2
-%yast_fillupdir /var/adm/fillup-templates
+%yast_libdir %{_libdir}/YaST2
+%yast_plugindir %{yast_libdir}/plugin
+%yast_includedir %{_includedir}/YaST2
+%yast_docdir %{_docdir}/%{name}
+%yast_logdir %{_localstatedir}/log/YaST2
+%yast_vardir %{_sharedstatedir}/YaST2
+%yast_fillupdir %{_localstatedir}/adm/fillup-templates
 
 %yast_execcompdir %{_prefix}/lib/YaST2
 %yast_agentdir %{yast_execcompdir}/servers_non_y2
 
-# build the yast module using autotools/make
-%yast_build \
-    %{_prefix}/bin/y2tool y2autoconf \
-    %{_prefix}/bin/y2tool y2automake \
-    autoreconf --force --install \
+# prepare sources for build
+%yast_prep \
+    %{_bindir}/y2tool y2autoconf \
+    %{_bindir}/y2tool y2automake \
+    \
+    # '-Wobsolete' exposes deprecated / obsolete Autotools-macros \
+    %{_bindir}/autoreconf \\\
+        --force \\\
+        --install \\\
+        -Wobsolete
+
+%yast_configure \
     \
     # '--disable-debug' adds '-DNDEBUG' to C(XX)FLAGS used by %%configure \
+    # '--disable-silent-rules'-switch overrides 'AM_SILENT_RULES(yes)' \
     %configure \\\
-        --disable-debug \
-    # V=1: verbose build in case we used AM_SILENT_RULES(yes) \
-    # so that RPM_OPT_FLAGS check works \
-    make %{?jobs:-j%jobs} V=1
+        --disable-debug \\\
+        --disable-silent-rules
+
+# build the yast module using autotools/make
+%yast_build \
+    \
+    # has %%yast_prep been run already? \
+    if [ ! -x "./configure" ]; then \
+        %yast_prep \
+    fi \
+    \
+    # has %%yast_configure been run already? \
+    if [ ! -f "./Makefile" ]; then \
+        %yast_configure \
+    fi \
+    \
+    %make_build
```

Comment:
`make_build` is in rpm-4.12.0 but openSUSE-13.2 still has 4.11.3. I have to check whether that matters.